src/Constraint/MatchAllConstraint.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of composer/semver.
+ *
+ * (c) Composer <https://github.com/composer>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace Composer\Semver\Constraint;
+
+/**
+ * Defines the absence of a constraint.
+ *
+ * This constraint matches everything.
+ */
+class MatchAllConstraint implements CompilableConstraintInterface
+{
+    /** @var string|null */
+    protected $prettyString;
+
+    /**
+     * @param ConstraintInterface $provider
+     *
+     * @return bool
+     */
+    public function matches(ConstraintInterface $provider)
+    {
+        return true;
+    }
+
+    public function compile($operator)
+    {
+        return 'true';
+    }
+
+    /**
+     * @param string|null $prettyString
+     */
+    public function setPrettyString($prettyString)
+    {
+        $this->prettyString = $prettyString;
+    }
+
+    /**
+     * @return string
+     */
+    public function getPrettyString()
+    {
+        if ($this->prettyString) {
+            return $this->prettyString;
+        }
+
+        return (string) $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function __toString()
+    {
+        return '[]';
```

Comment:
should we change this to `*` instead, to match how the VersionParser expects it to be represented ?

Reply:
Yup that makes sense, especially as we need a way to represent MatchNone probably for #100 - and `[]` might be a decent way to do that..